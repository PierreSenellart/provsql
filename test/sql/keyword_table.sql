\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

CREATE TABLE "order" (x INT, y INT);
INSERT INTO "order" VALUES (1,1),(1,2),(2,2),(3,3);

SELECT add_provenance('order');
SELECT create_provenance_mapping('select', 'order', 'x', 't');
CREATE TABLE result AS
  SELECT formula(provenance(),'select') FROM (SELECT DISTINCT 1 FROM "order");
SELECT remove_provenance('result');
SELECT * FROM result;
DROP TABLE result;
CREATE TABLE result AS
  SELECT sr_formula(provenance(),'select') FROM (SELECT DISTINCT 1 FROM "order");
SELECT remove_provenance('result');
SELECT * FROM result;
DROP TABLE result;

SELECT remove_provenance('order');
SELECT repair_key('order', 'x');
DO $$ BEGIN
  PERFORM set_prob(provenance(), (x+y)*.1) FROM "order";
END $$;
CREATE TABLE result AS
  SELECT x,probability_evaluate(provenance()) FROM (SELECT DISTINCT x FROM "order") ORDER BY x;
SELECT remove_provenance('result');
SELECT * FROM result;
DROP TABLE result;

DROP TABLE "order";
DROP TABLE "select";
