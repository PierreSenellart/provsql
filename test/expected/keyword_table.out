\set ECHO none
add_provenance

(1 row)
create_provenance_mapping

(1 row)
remove_provenance

(1 row)
formula
(1 ⊕ 1 ⊕ 2 ⊕ 3)
(1 row)
remove_provenance

(1 row)
sr_formula
(1 ⊕ 1 ⊕ 2 ⊕ 3)
(1 row)
remove_provenance

(1 row)
repair_key

(1 row)
remove_provenance

(1 row)
x|probability_evaluate
1|0.5
2|0.4
3|0.6
(3 rows)
